tests/src/SyncIntegrationTests.cpp

```diff
@@ -0,0 +1,157 @@
+        
+        std::vector<std::map<MessageId, Datum<double>>> result;
+        while (file.good()) {
+            std::map<MessageId, Datum<double>> data;
+            std::string line;
+            std::getline(file, line);
+            if (line.empty()) {
+                continue;
+            }
+            std::stringstream line_stream(line);
+            std::string cell;
+            std::getline(line_stream, cell, ',');
+            double file_time = std::stod(cell);
+            auto timestamp = now + milliseconds(static_cast<unsigned>(file_time * 1000));
+            
+            std::getline(line_stream, cell, ',');
+            Datum<double> power(std::stod(cell));
+            power.setTime(timestamp);
+            
+            std::getline(line_stream, cell, ',');
+            Datum<double> temperature(std::stod(cell));
+            temperature.setTime(timestamp);
+            
+            std::getline(line_stream, cell, ',');
+            Datum<double> voltage(std::stod(cell));
+            voltage.setTime(timestamp);
+            
+            data.insert({MessageId::Watts, power});
+            data.insert({MessageId::Centigrade, temperature});
+            data.insert({MessageId::Volts, voltage});
+            result.push_back(data);
+        }
+        return result;
+    }
+    
+    ConfigMap buildBaseMap() {
+        ConfigMap config;
+        config.set("model", "Battery");
+        config.set("predictor", "MC");
+        config.set("Predictor.loadEstimator", "Const");
+        config.set("Predictor.SampleCount", 100);
+        config.set("Predictor.Horizon", 10000);
+        config.set("Model.ProcessNoise", {"1e-5", "1e-5", "1e-5", "1e-5", "1e-5", "1e-5", "1e-5", "1e-5"});
+        config.set("LoadEstimator.Loading", 8);
+        return config;
+    }
+    
+    ConfigMap addUKFConfig(ConfigMap config) {
+        config.set("observer", "UKF");
+        config.set("Observer.Q", {"1e-10", "0", "0", "0", "0", "0", "0", "0", "0", "1e-10", "0", "0", "0", "0", "0", "0", "0", "0", "1e-10", "0", "0", "0", "0", "0", "0", "0", "0", "1e-10", "0", "0", "0", "0", "0", "0", "0", "0", "1e-10", "0", "0", "0", "0", "0", "0", "0", "0", "1e-10", "0", "0", "0", "0", "0", "0", "0", "0", "1e-10", "0", "0", "0", "0", "0", "0", "0", "0", "1e-10"});
+        config.set("Observer.R", {"1e-2", "0", "0", "1e-2"});
+        return config;
+    }
+    
+    ConfigMap addPFConfig(ConfigMap config) {
+        config.set("observer", "PF");
+        config.set("Observer.ParticleCount", "200");
+        config.set("Observer.ProcessNoise", {"1", "1", "1", "1", "1", "1", "1", "1"});
+        config.set("Observer.SensorNoise", {"1", "1"});
+        config.set("Observer.MinEffective", "100");
+        return config;
+    }
+    
+    void run_test(ConfigMap&& config) {
+        using namespace std::chrono;
+        // Read battery data from a file.
+        auto data = read_file("../../data/data_const_load.csv");
```

Comment:
Using CMake to copy the data file to the output directory is less likely to break than guessing the relationship between the output and source directories